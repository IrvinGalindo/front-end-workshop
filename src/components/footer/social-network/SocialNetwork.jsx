import React from 'react';

import PropTypes from 'prop-types';

const SocialNetwork = ({ site }) => (
  <li className="yta_Footer-socialNetwork-item">
    <a href="#">
      <span data-icon={`ei-sc-${site}`} data-size="m"></span>
      <span className="yta_u-onlyRead">{site}</span>
    </a>
  </li>
);

SocialNetwork.propTypes = {
  site: PropTypes.string.isRequired,
}

export default SocialNetwork;
