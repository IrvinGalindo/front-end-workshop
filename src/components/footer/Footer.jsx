import React from 'react';

import SocialNetwork from './social-network';

import './Footer.scss';

const socialNetworks = ['facebook', 'twitter', 'instagram', 'youtube'];

const Footer = () => (
  <footer className="yta_Footer">
    <h2 className="yta_u-onlyRead">Redes sociales</h2>
    <ul className="yta_Footer-socialNetwork">
      {
        socialNetworks.map(socialNetwork => <SocialNetwork key={socialNetwork} site={socialNetwork} />)
      }
    </ul>
    <p className="yta_Footer-copyright">Copyright Â© 2018 Yo te adopto.</p>
  </footer>
);

export default Footer;
