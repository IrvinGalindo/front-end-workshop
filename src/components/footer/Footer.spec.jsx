import React from 'react';
import { shallow } from 'enzyme';

import SocialNetwork from './social-network';
import Footer from './Footer';

describe('Footer', () => {
  const footer = shallow(<Footer />);

  it('should render', () => {
    expect(footer).toMatchSnapshot();
  });

  it('should render four <SocialNetwork />', () => {
    expect(footer.find(SocialNetwork).length).toBe(4);
  });
});
