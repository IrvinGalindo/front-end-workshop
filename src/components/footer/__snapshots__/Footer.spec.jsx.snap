// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<footer
  className="yta_Footer"
>
  <h2
    className="yta_u-onlyRead"
  >
    Redes sociales
  </h2>
  <ul
    className="yta_Footer-socialNetwork"
  >
    <SocialNetwork
      key="facebook"
      site="facebook"
    />
    <SocialNetwork
      key="twitter"
      site="twitter"
    />
    <SocialNetwork
      key="instagram"
      site="instagram"
    />
    <SocialNetwork
      key="youtube"
      site="youtube"
    />
  </ul>
  <p
    className="yta_Footer-copyright"
  >
    Copyright Â© 2018 Yo te adopto.
  </p>
</footer>
`;
